site/src/pages/DeploySettingsPage/LicensesSettingsPage/LicensesSettingsPageView.tsx

```diff
@@ -55,13 +60,29 @@ const LicensesSettingsPageView: FC<Props> = ({
           description="Manage licenses to unlock Enterprise features."
         />
 
-        <Button
-          component={Link}
-          to="/deployment/licenses/add"
-          startIcon={<AddIcon />}
-        >
-          Add a license
-        </Button>
+        <Stack direction="row" spacing={2}>
+          <Button
+            component={Link}
+            to="/deployment/licenses/add"
+            startIcon={<AddIcon />}
+          >
+            Add a license
+          </Button>
+          <Tooltip title="Refresh license entitlements. This is done automatically every 10 minutes.">
+            <Button
+              onClick={() => {
+                if (refreshEntitlements) {
+                  if (refreshEntitlements()) {
+                    displaySuccess("Successfully refreshed licenses")
```

Comment:
Sending an event is not a sync operation so does not make too much sense to display this message here. Since you are using XState for that, you could do:
- Call the `refreshEntitlements` function
- In the refresh button, add a prop `loading={...}`

Reply:
@BrunoQuaresma can I merge this and then we improve it?

Reply:
Yeap!